x-pack/plugins/fleet/server/routes/data_streams/handlers.ts

```diff
@@ -158,19 +194,17 @@ export const getListHandler: RequestHandler = async (context, request, response)
         // - and we didn't pick the metadata in an earlier iteration of this map()
         if (!packageMetadata[pkgName]) {
           // then pick the dashboards from the package saved object
-          const dashboards =
-            pkgSavedObject.attributes?.installed_kibana?.filter(
-              (o) => o.type === KibanaSavedObjectType.dashboard
-            ) || [];
-          // and then pick the human-readable titles from the dashboard saved objects
-          const enhancedDashboards = await getEnhancedDashboards(
-            context.core.savedObjects.client,
-            dashboards
-          );
+          const packageDashboardIds = dashboardIdsByPackageName[pkgName] || [];
+          const packageDashboards = packageDashboardIds.map((dashboardId) => {
```

Comment:
I believe using `reduce` here can avoid the `filter` a few lines later.

Reply:
thanks, fixed!